### PR TITLE
Collect libvirt VM metrics

### PR DESCRIPTION
**Pull request by usamasaqib**
### What does this PR do?
This PR introduces a collector to gather libvirt VM metrics in the metal instance, and push them to the datadog-agent via statsd.
The metrics collected are:
- Bytes swapped in from backing storage
- Bytes swapped out to backing storage
- Number of major page faults
- Memory available for a VM
- Memory balloon size for a VM
- RSS of VM process on host
- Unused memory by the system
- Usable memory without triggering a swap

### CODE REVIEW

**Review by pducolin — APPROVED** [review]

File: .gitlab/kernel_matrix_testing/common.yml [hunk 1]
```diff
@@ -169,6 +169,17 @@
...
+    - |
+      GO_ARCH=$ARCH
+      if [ "${ARCH}" == "x86_64" ]; then
+        GO_ARCH=amd64
+      fi
+    - cd test/new-e2e && GOOS=linux GOARCH="${GO_ARCH}" go build system-probe/vm-metrics/vm-metrics.go
+    # upload vm-metrics collector to metal instance
+    - scp -o StrictHostKeyChecking=no -i $AWS_EC2_SSH_KEY_FILE $CI_PROJECT_DIR/test/new-e2e/vm-metrics "ubuntu@$INSTANCE_IP:/home/ubuntu/vm-metrics"
+    # run vm-metrics collector
+    - ssh -o StrictHostKeyChecking=no -i $AWS_EC2_SSH_KEY_FILE "ubuntu@$INSTANCE_IP" "/home/ubuntu/vm-metrics -statsd-host=127.0.0.1 -statsd-port=8125 -libvirt-uri=/var/run/libvirt/libvirt-sock-ro --tag \"arch:${ARCH}\" --tag \"ci-pipeline-id:${CI_PIPELINE_ID}\" --daemon -log-file /home/ubuntu/daemon.log"
```
Comment (gjulianm) [inline .gitlab/kernel_matrix_testing/common.yml:182]: Shouldn't this be in the `.upload_{sysprobe,secagent}_tests` jobs, as the vm-metrics program is already prepared as a dependency there?

Comment (gjulianm) [inline .gitlab/kernel_matrix_testing/common.yml:182]: Add the test-component tag too to distinguish between sysprobe/secagent isntances

```suggestion
    - ssh -o StrictHostKeyChecking=no -i $AWS_EC2_SSH_KEY_FILE "ubuntu@$INSTANCE_IP" "/home/ubuntu/vm-metrics -statsd-host=127.0.0.1 -statsd-port=8125 -libvirt-uri=/var/run/libvirt/libvirt-sock-ro --tag \"arch:${ARCH}\" --tag \"test-component:${TEST_COMPONENT}\" --tag \"ci-pipeline-id:${CI_PIPELINE_ID}\" --daemon -log-file /home/ubuntu/daemon.log"
```

Comment (usamasaqib) [reply .gitlab/kernel_matrix_testing/common.yml:182]: If we place it in `.upload_{sysprobe,secagent}_tests`, then running it becomes complicated.
To run the vm-metrics collector, the `datadog-agent` must be running which happens in the `setup_env` stage. So we can run only after this stage is complete. So either in `setup_env` or `run_tests`.
If we run in `setup_env` the `upload_tests` stage may not have been completed in time. If we pick `run_tests`, we need to have some extra logic to ensure that we do not run the collector multiple times.

So the easiest options seems to be to just upload and run the collector in `setup_env`

Comment (gjulianm) [reply .gitlab/kernel_matrix_testing/common.yml:182]: Sounds good. I'd add a comment explaining why it needs to go there, just in case :D

**Review by gjulianm — APPROVED** [review]

**Pull request merged**